src/Nethermind/Nethermind.Core/Extensions/Bytes.cs

```diff
@@ -337,51 +337,75 @@ public static void Avx2Reverse256InPlace(Span<byte> bytes)
 
         public static uint ReadEthUInt32(this Span<byte> bytes)
         {
-            if (bytes.Length != 4)
+            if (bytes.Length > 4)
```

Comment:
W could abstract all those bytes manipulation out. The only thing those methods are different is using different Byteprimitives method and 4 vs 8 bytes

Reply:
how would you like to abstract them?

Reply:
`        public static uint ReadEthUInt32(this Span<byte> bytes) => BinaryPrimitives.ReadUInt32BigEndian(bytes.PadOrTrimToLastNBytes(4));

        public static ulong ReadEthUInt64(this Span<byte> bytes) => BinaryPrimitives.ReadUInt64BigEndian(bytes.PadOrTrimToLastNBytes(8));

        private static Span<byte> PadOrTrimToLastNBytes(this Span<byte> bytes, int n)
        {
            if (bytes.Length > n)
            {
                return bytes.Slice(bytes.Length - n, n);
            }
            else if (bytes.Length < n)
            {
                Span<byte> result = new byte[n];
                bytes.CopyTo(result.Slice(n - bytes.Length));
                return result;
            }
            else
            {
                return bytes;
            }
        }`

Didn't find a way to keep stackallock in though